### PR TITLE
Draft: Allow invoked pipelines to complete, mitigate blocking on status channel

### PR DESCRIPTION
**Pull request by peteski22**
**DRAFT: IN PROGRESS**

This PR is intended to be used initially for discussion of potential changes/improvements to `go-eventlogger`.

Perceived issues:

* `graph.process` The `Status` channel is changed from unbuffered (default size) to buffered, the size will be the estimated number of nodes within all pipelines (number of root nodes `* 3`). This is to prevent occurrence of sending messages to a blocked channel is no longer being received on
* `graph.process` While ranging over the root nodes, do not invoke roots when we know the context is already `Done`
* `graph.doProcess` parameter tweaked so the channel supplied can only be used for sending
* `broker.SetRequireAllPipelinesComplete` Added broker-level setting (per event type) to decide whether `process` should allow all invoked pipelines to complete and return their status before gathering/fanning-in

### CODE REVIEW

**Comment by peteski22** [comment]
**UPDATE**:

Most of this was a waste of time 😄 as `doProcess` doesn't allow returning the status if the context is cancelled. 

Side note: there was a bug in this anyway when we came to `Store` as we'd need to have checked if the pipeline already exists in the map and only increment if it didn't. 

TL;DR: Not really relevant and I had the wrong end of the stick, I should read the Go docs next time! 

`doProcess`: `// # No Status is sent when a request is cancelled by the context`
